### PR TITLE
More generic servicing

### PR DESCRIPTION
**Pull request by ericnewcomer**
Adds generic servicing for any view that uses `OrgObjPermsMixin` (and has a non org limited queryset). When encountered, users are redirected to a service view outside of the normal app framework. When they select `Service Account` they are then redirected back to the view they were trying to view. 

  * Remove perms from settings_common for `Customer Service`, not sure how django handles removing groups that people are in, but we no longer need this one.
  * Switch to `is_staff` instead of perms
  * Remove `Service` menu items

Screenshot 2023-01-30 at 5 53 51 PM

### Todo ###
- [x] Fix tests
- [x] Add coverage

### CODE REVIEW

File: static/js/urls.js [hunk 1]
```diff
@@ -43,23 +44,26 @@ window.urls = [
...
-          return path;
+    var match = path.match(mapping[findDirection]);
+    if (match) {
+      path = mapping[replaceDirection].source.replaceAll('\\/', '/');
+      for (var i = 1; i < match.length; i++) {      
+        path = path.replace('(.*)', match[i]);
       }
+      path = path.replaceAll('(.*)', '');
+      return path;
+    }
```
Comment (ericnewcomer) [inline static/js/urls.js:66]: Just fixing indentation here

File: temba/channels/tests.py [hunk 2]
```diff
@@ -500,8 +501,8 @@ def test_read(self):
 
         self.assertContentMenuContains(tel_channel_read_url, self.admin, "Enable Voice Calling")
 
-        two_hours_ago = timezone.now() - timedelta(hours=2)
-
+        test_date = datetime(2020, 1, 20, 0, 0, 0, 0, timezone.utc)
+        two_hours_ago = test_date - timedelta(hours=2)
```
Comment (ericnewcomer) [inline temba/channels/tests.py:505]: This was causing an intermittent failure based on when the test was run

File: temba/orgs/tests.py [hunk 3]
```diff
@@ -2883,7 +2883,7 @@ def test_home(self):
 
         self.login(self.admin)
 
-        with self.assertNumQueries(46):
+        with self.assertNumQueries(49):
```
Comment (ericnewcomer) [inline temba/orgs/tests.py:2886]: Extra org lookup per objorgperm view

File: temba/settings_common.py [hunk 4]
```diff
@@ -466,13 +466,7 @@
...
         "orgs.org_surveyor",
     ),
-    "Customer Support": (
-        "campaigns.campaign_read",  # anywhere we allow servicing still needs these
-        "channels.channel_read",
-        "channels.channellog_read",
-        "contacts.contact_read",
-        "flows.flow_editor",
-    ),
+    "Customer Support": (),
```
Comment (ericnewcomer) [inline temba/settings_common.py:469]: We don't need this group anymore, but haven't explored if there are side effects to removing it

File: templates/api/v2/api_explorer.haml [hunk 5]
```diff
@@ -151,7 +151,6 @@
 
       if (method == "GET") {
         store.getUrl(url+query, { "headers": headers }).then(function(response) {
-          console.log(response);
```
Comment (ericnewcomer) [inline templates/api/v2/api_explorer.haml:154]: Maybe this was intentional?

File: templates/frame.haml [hunk 6]
```diff
@@ -23,14 +23,21 @@
...
             var location = window.location;
-            var newUrl = mapUrl(location.pathname + (location.search || ""));
-            if (newUrl != window.location.pathname) {
-              window.location = newUrl;
-            } else { 
-              if (force) {
-                window.location = "/messages/";
+            var excluded = window.excludeUrls.find(function(url) {
+              return location.pathname.indexOf(url) > -1;
+            });
```
Comment (ericnewcomer) [inline templates/frame.haml:28]: To avoid ping-ponging on login, service, etc

File: templates/frame.haml [hunk 7]
```diff
@@ -87,6 +94,8 @@
       // npm packages
       %script{src:"{{ STATIC_URL }}intercooler/dist/intercooler.js"}
 
+      %script(src="{{ STATIC_URL }}rest_framework/js/prettify-min.js")
```
Comment (ericnewcomer) [inline templates/frame.haml:97]: We use prettify all over, so moving it up.

File: templates/spa_frame.haml [hunk 8]
```diff
@@ -817,15 +825,17 @@
         showLoading(true);
         var store = document.querySelector("temba-store");
         store.postUrl("/org/choose/", "organization=" + orgId, {}, 'application/x-www-form-urlencoded').then(function(response) {
-          document.location.href = "/";
+          if (response.redirected) {
+            document.location.href = response.url;
+          }
```
Comment (ericnewcomer) [inline templates/spa_frame.haml:830]: This makes sure we honor the response location from choose which should route to a role appropriate page after the switch

File: templates/spa_frame.haml [hunk 9]
```diff
@@ -854,7 +864,7 @@
 
   .flex-col
     .flex.widget-container(style="height:100vh;overflow:hidden")
-      -if not request.user.is_staff
+      -if request.user.is_staff
```
Comment (ericnewcomer) [inline templates/spa_frame.haml:867]: ooopsie

File: templates/request_logs/httplog_read.haml [hunk 10]
```diff
@@ -42,11 +42,6 @@
 
 -block extra-script
   {{ block.super }}
-  <script src="{{ STATIC_URL }}rest_framework/js/prettify-min.js"></script>
-  :javascript
-    $(function(){
-      prettyPrint();
-    });
```
Comment (rowanseymour) [inline templates/request_logs/httplog_read.haml:49]: hi @ericnewcomer

**Review by rowanseymour — APPROVED** [review]

File: templates/request_logs/httplog_read.haml (hunk 10 above)
Comment (ericnewcomer) [reply templates/request_logs/httplog_read.haml:49]: hi @rowanseymour!

**Pull request merged**